typha/pkg/syncserver/sync_server.go

```diff
@@ -566,6 +555,7 @@ func (h *connection) handle(finishedWG *sync.WaitGroup) (err error) {
 	if err = h.doHandshake(); err != nil {
 		return // Error already logged.
 	}
+	h.gaugeNumConnections.Inc()
```

Comment:
Do we need a nil check here as well? I see the Dec() call above is protected.

Reply:
I suppose moving this down below the handshake changes this gauge's meaning as well, but that's probably OK.

Reply:
No, the gauge is set up by `doHandshake()` so no nil check needed but the defer block runs even if we don't get this far.  Suppose I could do `defer h.gaugeNumConnections.Dec()` here instead but previously we only decremented after all connection resources were freed.

Reply:
Hmm, some tests were broken by that change (and they were using it to test the interaction with handshake).  I'll reinstate the old stat and add a new one for each syncer type.